internal/executor/instance/instance.go

```diff
@@ -126,12 +133,42 @@ func (inst *Instance) Run(ctx context.Context, config *RunConfig) error {
 			Memory:   int64(inst.memory * mebi),
 		},
 	}
+
+	// In case the additional containers are used, tell the agent to wait for them
+	if len(inst.additionalContainers) > 0 {
+		var ports []string
+		for _, additionalContainer := range inst.additionalContainers {
+			ports = append(ports, strconv.FormatUint(uint64(additionalContainer.ContainerPort), 10))
+		}
+		commaDelimitedPorts := strings.Join(ports, ",")
+		containerConfig.Env = append(containerConfig.Env, "CIRRUS_PORTS_WAIT_FOR="+commaDelimitedPorts)
+	}
+
 	cont, err := cli.ContainerCreate(ctx, &containerConfig, &hostConfig, nil, "")
 	if err != nil {
 		return err
 	}
 
+	// Start additional containers (if any)
+	var acWG sync.WaitGroup
+	acErrChan := make(chan error, len(inst.additionalContainers))
+	acCtx, acCancel := context.WithCancel(context.Background())
```

Comment:
Should we use `ctx` here so parent cancelation is handeled too?

```suggestion
	acCtx, acCancel := context.WithCancel(ctx)
```

Reply:
It's handled through `acCancel()` in `defer`, which we'll need to call anyway in case of successful termination.

Another reason is that using a separate context here makes cancellation order more predictable, which helps when reasoning about `Instance.Run()`'s properties.

Reply:
I don't see how `Interrupt` will be handled in this case. You've added `ctx` for that reason, don't you?

Reply:
> I don't see how `Interrupt` will be handled in this case.

https://github.com/cirruslabs/cirrus-cli/blob/f0d70b442879f54bfcae1e3c1b94fde9a1c34b4e/internal/executor/instance/instance.go#L168-L169

Reply:
I don't believe there is any guarantees that a `defer`ed function will be called on cancelation.

Reply:
So the `ctx` will get canceled, then `errChan` from `ContainerWait` will emit an error which then return from the function and finally the `defer`ed function will be called.

Reply:
Let' add a comment then: `// separate context so when 'ctx' is canceled the main container will be killed first and only then all additional containers will be killed too`

Reply:
See https://github.com/cirruslabs/cirrus-cli/pull/15/commits/1e7c3b93b9211257d4d5f91943b06d6b45ca0c22.